pkg/R/sparkR.R

```diff
@@ -6,14 +6,26 @@ sparkR.onLoad <- function(libname, pkgname) {
   assemblyJarPath <- paste(libname, "/SparkR/", assemblyJarName, sep="")
   assemblyJarPath <- gsub(" ", "\\ ", assemblyJarPath, fixed=T)
   packageStartupMessage("[SparkR] Initializing with classpath ", assemblyJarPath, "\n")
-
-  sparkMem <- Sys.getenv("SPARK_MEM", "512m")
-  yarn_conf_dir <- Sys.getenv("YARN_CONF_DIR", "")
-  
+ 
   .sparkREnv$libname <- libname
   .sparkREnv$assemblyJarPath <- assemblyJarPath
-  .jinit(classpath=assemblyJarPath, parameters=paste("-Xmx", sparkMem, sep=""))
-  .jaddClassPath(yarn_conf_dir)
+}
+
+#' Stop this Spark context
+#' Also terminates the backend this R session is connected to
+sparkR.stop <- function(sparkREnv) {
+  if (exists(".sparkRjsc", envir=.sparkREnv)) {
+    sc <- get(".sparkRjsc", envir=.sparkREnv)
+    callJMethod(sc, "stop")
+  }
+
+  if (exists(".sparkRCon", envir=.sparkREnv)) {
+    callJStatic("SparkRHandler", "stopBackend")
+    # Also close the connection and remove it from our env
+    conn <- get(".sparkRCon", .sparkREnv)
+    close(conn)
+    rm(".sparkRCon", envir=.sparkREnv)
```

Comment:
Well technically this could be called whenever `sc` goes out of scope and not just on exit. So I'm going to leave this in just to be safe.